src/main/java/com/commercetools/sync/cartdiscounts/CartDiscountSync.java

```diff
@@ -105,12 +105,14 @@ private void handleError(@Nonnull final String errorMessage,
 
     private String getKey(@Nonnull final CartDiscount cartDiscount) {
         //todo: SUPPORT-4443 need to be merged from name to key.
-        return cartDiscount.getName().get(Locale.ENGLISH);
+        return Optional.ofNullable(cartDiscount.getName())
+                       .map(localizedString -> localizedString.get(Locale.ENGLISH)).orElse("");
```

Comment:
Why is English locale of name used as key? Why not other locale?

Reply:
good question, I will try to explain it.. this is the default locale for a name field if you open an empty project in CTP.  So I prefer to check eng version as a key. It's a temporary solution because the CTP does not have a key (it's developing now and will be released soon) until CTP has this key field, we are agreed with this option as a unique key field for sync operations.